### PR TITLE
Use GitHub Discussions with report suggestion

### PR DESCRIPTION
**Pull request by hsbt**
## What was the end-user or developer problem that led to this PR?

`Bundler::FriendlyErrors` still suggests to report our issues. But their report is not helpful for developpers and mostly there is no response from first reporter. We should switch this template to discussions from issues. If we decide to handle it as issues, we can easily convert it to issues.

## What is your fix for the problem, implemented in this PR?

replaced GH issues to GH discussions.

## Make sure the following tasks are checked

- [x] Describe the problem / feature
- [x] Write [tests](https://github.com/rubygems/rubygems/blob/master/bundler/doc/development/PULL_REQUESTS.md#tests) for features and bug fixes
- [x] Write code to solve the problem
- [x] Make sure you follow the [current code style](https://github.com/rubygems/rubygems/blob/master/bundler/doc/development/PULL_REQUESTS.md#code-formatting) and [write meaningful commit messages without tags](https://github.com/rubygems/rubygems/blob/master/bundler/doc/development/PULL_REQUESTS.md#commit-messages)

### CODE REVIEW

File: bundler/lib/bundler/friendly_errors.rb
Comment (duckinator) [inline bundler/lib/bundler/friendly_errors.rb:80]: Maybe this?:

```suggestion
        First, try this link to see if anyone has reported this error:
```

File: bundler/lib/bundler/friendly_errors.rb
Comment (duckinator) [inline bundler/lib/bundler/friendly_errors.rb:110]: Given the link now creates a new Discussion, not a new Issue, we should update this function name accordingly.

```suggestion
    def new_discussion_url
```

File: bundler/lib/bundler/friendly_errors.rb
Comment (duckinator) [inline bundler/lib/bundler/friendly_errors.rb:83]: This changes `new_issue_url` to `new_discussion_url`, to match my other suggestion.

```suggestion
        If there aren't any reports for this error yet, please fill in the new issue form located at #{new_discussion_url}, and copy and paste the report template above in there.
```

**Review by duckinator — CHANGES_REQUESTED** [review]
I don't have strong feelings other way about this change, but _if_ this does get merged I'd like the message to be simplified a bit and for `new_issue_url` to be renamed to `new_discussion_url`.

Specifically, I think this phrasing is a bit easier to understand:

> First, try this link to see if anyone has reported this error:

I provided these as "suggested changes" you can apply directly from GitHub, but feel free to make the changes locally yourself if you prefer! 🙂

**Comment by hsbt** [comment]
@duckinator Thank you for your review. Just update them.

**Review by duckinator — APPROVED** [review]
Looks good to me.

**Comment by deivid-rodriguez** [comment]
I disagree with this. Whenever the bug report template is printed to end users, that's a bug in Bundler and it's helpful to get it reported. We still get bad quality reports where users only copy the template without providing any more information, but I don't think recommending opening a discussion improves that.

**Comment by deivid-rodriguez** [comment]
I think I've never converted a "Bug report template" issue into a discussion? 🤔

**Comment by hsbt** [comment]
It's not for improving report quality. Discussions is helpful with community support. In fact, https://github.com/rbenv/ruby-build/discussions is handled by many users.

**Comment by deivid-rodriguez** [comment]
As I explained I'm against this because bug reports should get reported to the bug tracker, not as discussion. I'm happy to get wider community help with triaging issues, but the right place for that is the issue tracker.

**Comment by hsbt** [comment]
I don't think so.

ok, I withdraw this because I didn't get your approval.

**Comment by deivid-rodriguez** [comment]
I'm sorry! I'm not the only maintainer, if other maintainers prefer this way I can accept the change. I can also triage discussions so it's not a big deal. We can reopen if there's more support for this change.

**Comment by hsbt** [comment]
👌 @indirect @simi How about this? 

https://github.com/rubygems/rubygems/discussions/6760 and https://github.com/rubygems/rubygems/discussions/7183 are good example for this PR.

**Comment by indirect** [comment]
I’m ok with either one, since we have to triage either way.

I think for common copy and paste errors we should change Bundler to not suggest an error report. 😅 For example, if there is a permission error we could print “Bundler cannot continue because it does not have permission to write to /path.” instead of suggesting a new issue/discussion.

**Comment by deivid-rodriguez** [comment]
Absolutely!

That's exactly the direction I've been taking for a while for addressing these issues. See for example https://github.com/rubygems/rubygems/pull/7113, https://github.com/rubygems/rubygems/pull/5726, https://github.com/rubygems/rubygems/pull/4965.

My take is that the bug report being printed is always a Bundler bug, because either:
* It shows a real bug in Bundler and provides helpful info to address it.
* It does NOT show a real bug in Bundler, but an expected error condition. In that case, printing the bug report template is also a bug, since we should be printing instead a proper informative error to the user and not the bug report template.

**Comment by hsbt** [comment]
okay, I'll reopen this when we will get many of not helpful issues.

**Pull request closed without merging**
